### ISSUE TITLE
update-child-modules is doing more than I wish...

### ISSUE THREAD

**Issue by chris-nerr** [mojohaus/versions-maven-plugin#190]
Hello!
I have a maven project with submodules:

A 1.0.0
- B 1.1.0
- - B-one 1.1.0
- - B-two 1.1.0
- - B-three 1.1.0
- C 1.1.1
- - C-one 1.1.1
- - C-two 1.1.1

All sub-modules of level 3 have the same version than its parent, for example version of module **B-two** is the same than its parent **B**.
If I change the version of my root project **A** to 2.0.0 and I run the command `mvn versions:update-child-modules -DallowSnapshots=true` from the root project then the version of the parent of the 3rd-level modules is updated: version of the C-one parent becomes 2.0.0!!
But that is not what I wish, I just want to update the version of artifact A and its direct children = modules of 2nd level.
Help! :)
Thanks. Chris.

**Comment by khmarbaise** [comment]
Taken a look on http://www.mojohaus.org/versions-maven-plugin/update-parent-mojo.html ? Best would be you make an example project on Github and describe what you've done and then what you like to see...a concrete project is better than describing it in a long abstract...

**Comment by DiJu519** [comment]
I understand the issue, mainly because i'm hitting it as well

Say I have 3 layers of parent inheritance: A, B, and C

A will be an aggregate pom

 A pom.xml

```
  <groupId>com</groupId>
    <artifactId>A</artifactId>
    <version>3.0.0</version>
    <modules>
       <module>B1</module>
        <module>B2</module>
        <module>B3</module>
    </modules>
```

B1 pom.xml
 
```
  <parent>
        <groupId>com</groupId>
    <artifactId>A</artifactId>
    <version>3.0.0-SNAPSHOT</version>
    </parent>

  <groupId>com</groupId>
     <artifactId>B1</artifactId>
     <version>1.0.0-SNAPSHOT</version>

    <modules>
         <module>C1</module>
        <module>C2</module>
         <module>C3</module>
     </modules>
```


 
 C1 pom.xml
 
```
     <parent>
         <groupId>com</groupId>
   <artifactId>B1</artifactId>
    <version>1.0.0-SNAPSHOT</version>
     </parent>
```


Now if i  execute this goal on the A pom.xml, not only will it update the Parent of B1 pom.xml (which is good, because B1's parent is A, and makes sense since the Group and Artifact match

BUT 

it will also update the parent of C1's pom.xml, which is BAD, as C1's parent should not jump to 3.0.0, as C1's parent is not A, it's B1.

Basically, this goal is blindly setting all parents to whatever is in A, instead of actually checking if a submodules parent is that of what's in the aggregate pom.

**Comment by DiJu519** [comment]
I think i found the issue, i also have some additional unit tests created.

I'll attempt to push them up for review. Need to get my company's OK first however.

**Comment by khmarbaise** [comment]
@DiJu519 Do you have the Ok ?

**Comment by DiJu519** [comment]
Still waiting.

**Comment by khmarbaise** [comment]
Ok. Thanks for the fast feedback.

**Comment by rbeckert** [comment]
See: https://github.com/mojohaus/versions-maven-plugin/issues/312#issuecomment-472498993
